src/views/components/AnalyticsGraph/index.vue

```diff
@@ -0,0 +1,312 @@
+<template>
+  <div :id="this.title">
+    <svg width="80vw" height='33vh'></svg>
+  </div>
+</template>
+<script>
+import * as d3 from "d3";
+export default {
+  name: 'graph',
+  data () {
+    return {
+    };
+  },
+  props : ['values' , 'title', 'xUnits', 'yUnits'],
+
+  mounted() {
+    /**@description Create an event listener for window resizing that rerenders the graph */
+    this.$nextTick(function() {
+      window.addEventListener('resize', this.redrawGraph);
+    })
+    //initialize the graph
+      this.drawLinesGraph(document.getElementById(this.title).offsetHeight * 0.90, document.getElementById(this.title).offsetWidth * 0.85, 
+      this.values, this.title, this.xUnits, this.yUnits);
+  },
+  beforeDestroy() {
+    window.removeEventListener('resize', this.redrawGraph);
+  },
+  methods : {
+    /** @description remove the specified SVG and rerender it with the new window dimensions */
+    redrawGraph() {
+      d3.select('#' + this.title).selectAll('g').remove();
+      this.drawLinesGraph(document.getElementById(this.title).offsetHeight * 0.90, document.getElementById(this.title).offsetWidth * 0.85, 
+      this.values, this.title, this.xUnits, this.yUnits);
```

Comment:
Try running your formatter in vscode! maybe it'll format these function calls so theyre easier to read

Reply:
I tried :( it isn't working in vue files for some reason I might need a vue extension or something